impeller/geometry/path.h

```diff
@@ -90,10 +90,20 @@ class Path {
 
   /// One or more contours represented as a series of points and indices in
   /// the point vector representing the start of a new contour.
+  ///
+  /// Polylines are ephemeral and meant to be used by the tessellator. They do
+  /// not allocate their own point vectors to allow for optimizations around
+  /// allocation and reuse of arenas.
   struct Polyline {
+    /// The buffer must remain valid for the lifetime of this object.
+    explicit Polyline(std::vector<Point>& point_buffer);
+
     /// Points in the polyline, which may represent multiple contours specified
     /// by indices in |breaks|.
-    std::vector<Point> points;
+    std::vector<Point>& points;
```

Comment:
This makes me a bit nervous, storing a pointer as an instance method.  Potentially you could instead recycle these and have a unique_ptr around them.  So you pass it ownership when you create it.  Then you take ownership, then you can reuse it with the next Polyline if you want.

Reply:
I reworked the patch to use unique_ptr and have a callback fired in the dtor to reclaim it for code paths that need it. I think this cleans up the API a little bit, and it keeps the benchmark happy.